python/plugins/MetaSearch/dialogs/newconnectiondialog.py

```diff
@@ -94,8 +94,14 @@ def accept(self):
             self.settings.setValue(keyurl, conn_url)
             self.settings.setValue('/MetaSearch/selected', conn_name)
 
-            self.settings.setValue('%s/username' % key, conn_username)
-            self.settings.setValue('%s/password' % key, conn_password)
+            if conn_username != '':
```

Comment:
Given the `...or None` additions in `python/plugins/MetaSearch/dialogs/maindialog.py`, we could make this a bit more robust with:

```python
if conn_username not in ['', None]:
```

Reply:
It seems to me, at that point in the code, conn_username and conn_password can only be either a non empty string or an empty string (given that `conn_username = self.leUsername.text().strip()` and `conn_password = self.lePassword.text().strip()`), but not `None`.

The `... or None` is needed to avoid to pass an empty string as username and password to get_catalog_service and passing a `None` value instead.

Do you thing it's better to leave the  `... or None` in the call to get_catalog_service in maindialog, or put them directly in get_catalog_service or in CSW202Search?